src/periods/periods.service.ts

```diff
@@ -28,6 +28,18 @@ export class PeriodsService {
     }
   }
 
+  async search(query: CreatePeriodDto): Promise<Period[]> {
```

Comment:
the search param must be included into the get all search

Reply:
async findAll(query: CreatePeriodDto): Promise<PaginatedResponse<Period>> {
    const { page = 1, size = 10, search } = query;
    const { take, skip } = createPaginationMetadata(page, size);
    const prismaQuery = {
      take,
      skip,
      where: search ? { name: { contains: search } } : {},
    };
    const [period, total] = await Promise.all([
      this.prismaService.period.findMany(prismaQuery),
      this.prismaService.period.count(),
    ]);
    return createPaginatedResponse<Period>(period, total, page, size);
  }